process_report/processors/coldfront_fetch_processor.py

```diff
@@ -0,0 +1,104 @@
+import os
+import functools
+import logging
+from dataclasses import dataclass
+
+import requests
+
+from process_report.invoices import invoice
+from process_report.processors import processor
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+CF_ATTR_ALLOCATED_PROJECT_NAME = "Allocated Project Name"
+
+
+@dataclass
+class ColdfrontFetchProcessor(processor.Processor):
+    nonbillable_projects: list[str]
+
+    @functools.cached_property
+    def coldfront_client(self):
+        keycloak_url = os.environ.get("KEYCLOAK_URL", "https://keycloak.mss.mghpcc.org")
+
+        # Authenticate with Keycloak
+        token_url = f"{keycloak_url}/auth/realms/mss/protocol/openid-connect/token"
+        try:
+            r = requests.post(
+                token_url,
+                data={"grant_type": "client_credentials"},
+                auth=requests.auth.HTTPBasicAuth(
+                    os.environ["KEYCLOAK_CLIENT_ID"],
+                    os.environ["KEYCLOAK_CLIENT_SECRET"],
+                ),
+            )
+        except KeyError:
+            logger.error(
+                "Keycloak authentication failed. Please set the environment variables KEYCLOAK_CLIENT_ID and KEYCLOAK_CLIENT_ID"
+            )
```

Comment:
You log an error here, but then just keep moving forward. Is that intentional, or should you abort the operation here?

Reply:
If these environment variables are *required*, should we be checking for them at an earlier point in program execution?

Reply:
I will add a function in `process_report.process_report.py` that will check for all required env vars. I will have this function called at the start of `main()` before the s3 input files are fetched.